experimental/stats/metricregistry.go

```diff
@@ -0,0 +1,240 @@
+/*
+ *
+ * Copyright 2024 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package stats
+
+import (
+	"log"
+
+	"google.golang.org/grpc/stats"
+)
+
+// DefaultMetrics are the default metrics registered through global metrics
+// registry. This is written to at initialization time only, and is read only
+// after initialization.
+var DefaultMetrics = stats.NewMetrics()
+
+// MetricDescriptor is the data for a registered metric.
+type MetricDescriptor struct {
+	// Name is the name of this metric. This name must be unique across whole
+	// binary (including any per call metrics). See
+	// https://github.com/grpc/proposal/blob/master/A79-non-per-call-metrics-architecture.md#metric-instrument-naming-conventions
+	// for metric naming conventions.
+	Name stats.Metric
+	// Description is the description of this metric.
+	Description string
+	// Unit is the unit of this metric (e.g. entries, milliseconds).
+	Unit string
+	// Labels are the required label keys for this metric. These are intended to
+	// metrics emitted from a stats handler.
+	Labels []string
+	// OptionalLabels are the optional label keys for this metric. These are
+	// intended to attached to metrics emitted from a stats handler if
+	// configured.
+	OptionalLabels []string
+	// Default is whether this metric is on by default.
+	Default bool
+	// Type is the type of metric. This is set by the metric registry, and not
+	// intended to be set by a component registering a metric.
+	Type MetricType
+}
+
+// Int64CountHandle is a typed handle for a float count metric. This handle
+// is passed at the recording point in order to know which metric to record
+// on.
+type Int64CountHandle struct {
+	MetricDescriptor *MetricDescriptor
+}
+
+// Record records the int64 count value on the metrics recorder provided.
+func (h Int64CountHandle) Record(recorder MetricsRecorder, incr int64, labels ...string) {
```

Comment:
Should these just be passed around by pointer everywhere?  Why not?

Reply:
Sure we could. This is a copy of a that x bit pointer but pointer works too.